grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -596,15 +596,19 @@ class Product implements Comparable, Serializable {
         }
     }
 
-    def validateRequiredFields() {
+    Boolean validateRequiredFields() {
+        return this.validateRequiredFields([])
+    }
+
+    Boolean validateRequiredFields(List<ProductField> ignoreProductFields) {
```

Comment:
Did you try to assign an empty array as a default value instead of creating a second method?
I mean:
```
Boolean validateRequiredFields(List<ProductField> ignoreProductFields = []) {
```
And now when you call this method without a passed argument, the empty array will be used.

Reply:
I did not try this, I just followed the conversation that we usually do for these types of cases in the project